nodejs/week1/meals/src/backend/routes/cheap-meals.js

```diff
@@ -0,0 +1,17 @@
+const express = require("express");
+const app = express();
+const meals = require("../data/meals.json");
+const reviews = require("../data/reviews.json");
+
+app.get("/cheap-meals", (req, res) => {
+  const cheapMeals = meals
+    .filter((meal) => meal.price < 75)
+    .map((meal) => {
+      meal.reviews = reviews.filter((review) => meal.id === review.mealId);
+      return meal;
+    });
+
+  res.send(cheapMeals);
+});
```

Comment:
This is a great use of map and filter.

Hint: This could be extracted into a function that gives you mealsAndReviews. Consider putting this function in its own file and using it in all other routers that need meals with reviews.

In general, try to break down problems into their small parts:

e.g. Get cheap meals with reviews
Step 1) Get meals with reviews
Step 2) Get **cheap** meals with reviews

When you break down problems like this, you'll find that step 1) is repeated across many other problems:

* Get larger meals (including reviews)
Step 1) Get meals with reviews
Step 2) Get _larger_ meals with reviews
* Get random meal (including reviews)
Step 1) Get meals with reviews
Step 2) Get one random meal with reviews

When you find yourself writing the same code more than twice, consider whether you can extract a helper function that you can reuse from both places. This principle is commonly known as DRY (Don't Repeat Yourself). Keeping your code DRY helps you right more maintainable code because:

* If there's a bug in this code, it's only in one place and you only need to fix it once
* When you create a helper function other team mates can benefit from it and reuse your code
* When someone looks at your code they need to read it slowly and understand what is going on. When they read a helper function like getRandomInteger(maxValue) they automatically understand what the function does without needing to exert their cognitive abilities a lot.

Reply:
Thank you for giving a very detailed feedback. I have made the changes by adding  a separate function.